test/unit/constraints/array.test.ts

```diff
@@ -0,0 +1,113 @@
+import "../metadataShim";
+import {registerJoi, WORKING_SCHEMA_KEY} from "../../../src/core";
+import * as Joi from 'joi';
+import { Max, Length, ArraySchema, Min, Unique } from "../../../src/constraints/array";
+import {Validator} from "../../../src/Validator";
+import { testConstraint } from "../testUtil";
+
+
+registerJoi(Joi);
+
+describe('Array Test', () => {
+    describe('Test Array methods ', () => {
+        it('Should annotate class', () => {
+            class MyClass {
+                @ArraySchema()
+                myProperty!: Array<number>
+            }
+            const metadata = Reflect.getMetadata(WORKING_SCHEMA_KEY, MyClass.prototype);
+            const expected = {
+                myProperty: Joi.array()
+            }
+            expect(metadata).toEqual(expected)
+        })
+        it('Should allow correct max value for array', () => {
+            class MyArrayClass {
+                @Max(5)
+                @ArraySchema()
+                myArray!: Array<number>
+            }
+            let object = new MyArrayClass();
+            object.myArray = [3,2,3,3];
+            const validator = new Validator();
+            const result = validator.validateAsClass(object, MyArrayClass);
+            expect(result.error).toBeNull();
+        })
+        it('Should allow correct max value for array - Negative Test', () => {
```

Comment:
⛏ Let's keep a blank line between `it()` statements, to help visually separate each test case.